agent/engine/engine_integ_test.go

```diff
@@ -169,6 +170,11 @@ func TestHostVolumeMount(t *testing.T) {
 	data, err := ioutil.ReadFile(filepath.Join(tmpPath, "hello-from-container"))
 	assert.Nil(t, err, "Unexpected error")
 	assert.Equal(t, "hi", strings.TrimSpace(string(data)), "Incorrect file contents")
+
+	client := taskEngine.(*DockerTaskEngine).client
```

Comment:
What do you think about putting the cleanup code in a `defer` block above the main code and verifications?

Same for the other Tests.

Reply:
Good idea! Will incorporate

Reply:
using a defer is not as appropriate in this situation because a defer is used to clean up the task engine. We can factor out the code and place it at the end of each method tho.